src/cdk-experimental/menu/menu-trigger.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, Injector} from '@angular/core';
+import {Menu} from './menu-interface';
+import {MenuStack} from './menu-stack';
+
+@Injectable()
+export class MenuTrigger {
+  private _childMenuInjector?: Injector;
+
+  protected childMenu?: Menu;
+
+  constructor(protected injector: Injector, protected menuStack: MenuStack) {}
+
+  protected getChildMenuInjector() {
+    this._childMenuInjector =
+      this._childMenuInjector ||
+      Injector.create({
+        providers: [
+          {provide: MenuTrigger, useValue: this},
+          {provide: MenuStack, useValue: this.menuStack},
+        ],
+        parent: this.injector,
```

Comment:
I think that the `parent` here isn't necessary since the embedded view injector is inserted in the node injector tree.

Reply:
This does seem to make a difference. If I remove this then submenus don't automatically expand when I hover over a trigger item. I'm not sure exactly why that happens, but there must be something important passed along in the injector context